PDO

globeFrEak opened the pull request:
- PDO Driver hinzugefügt (includes/pdo_database_driver.php)
- alte DB Funktionen aus maincore.php und setup.php entfernt
- pdo_database_driver.php in maincore.php und setup.php eingebunden

globeFrEak committed:
Add PDO Driver

added includes/pdo-database-driver.php to maincore.php

delete old DB-Functions from maincore.php

globeFrEak committed:
Setup PDO

add PDO to setup.php and remove old functions

globeFrEak committed:
PHPDoc

PHPDoc lines added for PDO functions

globeFrEak committed:
PHPDoc #2

globeFrEak committed:
file description

iPaat commented:
Sehr gute Arbeit, 
vielen Dank :+1:

Webmeteor commented:
Alle Dateien, wo ein INPUT Query vorkommt müssen nun noch überprüft 
werden, wegen mysql_insert_id, da diese Funktion ja nicht mehr 
funktionieren wird.

Am 27.06.2016 um 20:44 schrieb Patrick:

> Sehr gute Arbeit,
> vielen Dank 👍
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> https://github.com/PHP-Fusion-GERMANY/PHP-Fusion/pull/19#issuecomment-228837723, 
> or mute the thread 
> https://github.com/notifications/unsubscribe/AMRmIUotQnx8E6CDvhx0b00dRvOU4JRNks5qQBongaJpZM4I_Yl9.

---

Diese E-Mail wurde von Avast Antivirus-Software auf Viren geprüft.
https://www.avast.com/antivirus

globeFrEak committed:
mysql_insert_id -< dblastid

globeFrEak commented:
Ja stimmt, habe ich geändert

Danke

iPaat commented:
https://github.com/PHP-Fusion-GERMANY/PHP-Fusion/blob/develop/administration/db_backup.php

enthält noch mysql funktionen

globeFrEak committed:
PDO Attribute uncomm

globeFrEak committed:
DB Connection

change MySQL statements into PDO statements

globeFrEak committed:
#1 PDO for Backup

not final

globeFrEak commented:
Ich brauche Hilfe bei `mysql_unbuffered_query($result);`, dazu müsste man bei PDO ein extra Attribute bei der Verbindung setzen - wie machen wir das am besten?
https://github.com/PHP-Fusion-GERMANY/PHP-Fusion/pull/19/commits/eaac73d129bfa07af63252c0ca223b7af51b3d6e

---

Ansonsten habe ich noch alle `mysql_close` statements ersetzt und die DB-Verbindung in $db_connect gepackt - müssen wir die Verbindung schließen bei PDO?!
https://github.com/PHP-Fusion-GERMANY/PHP-Fusion/pull/19/commits/afd9043dcfc341b7eb1d6fd7d924cdef0cddfb58

iPaat commented:
Entweder 2 Funktionen:

``` php
function dbSetUnbuffered() {
    global $pdo;
    $pdo->setAttribute(PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, false);
}

function dbRevertUnbuffered() {
    global $pdo;
    $pdo->setAttribute(PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, true);
}
```

Dann müsste der Code lauten:

``` php
dbSetUnbuffered();
dbquery($result);
dbRevertUnbuffered();
```

---

Oder in einer Funktion:

``` php
function dbUnbufferedQuery($query, $execute = array()) {
    global $pdo;

    $pdo->setAttribute(PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, false);
    $ret = dbquery($query, $execute = array());
    $pdo->setAttribute(PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, true);

    return $ret;
}
```

Dann müsste der Code lauten:

``` php
dbUnbufferedQuery($result);
```

---

Die Verbindung wird geschlossen wenn man `$pdo = null;` setzt.

globeFrEak committed:
PDO Close #2

globeFrEak committed:
PDO for setup.php (complete)

globeFrEak committed:
PDO Exceptions from old columns

user_threads and user_post exist not without forum

globeFrEak committed:
add dbUnbufferedQuery

wirft aber noch eine Exception:
Fatal error: Uncaught exception 'PDOException' with message
'SQLSTATE[42S22]: Column not found: 1054 Unknown column '' in 'field
list'' in
C:\xampp\htdocs\github\PHP-Fusion\includes\pdo_database_driver.php:59
Stack trace: #0
C:\xampp\htdocs\github\PHP-Fusion\includes\pdo_database_driver.php(59):
PDO->prepare('\n\n#\n# Table Dat...') #1
C:\xampp\htdocs\github\PHP-Fusion\includes\pdo_database_driver.php(203):
dbquery('\n\n#\n# Table Dat...', Array) #2
C:\xampp\htdocs\github\PHP-Fusion\administration\db_backup.php(167):
dbUnbufferedQuery('\n\n#\n# Table Dat...') #3 {main} thrown in
C:\xampp\htdocs\github\PHP-Fusion\includes\pdo_database_driver.php on
line 59

Hier scheint was mit dem PREG nicht zu stimmen, leider finde ich nicht
warum das nun auf einmal falsch sein soll.

Bsp: var_dump von result = string(126) " # # Table Data for
`fusion_smileys` # INSERT INTO `fusion_smileys` (``, ``, ``, ``) VALUES
('1', ':)', 'smile.gif', 'Smile')"

globeFrEak commented:
wirft aber noch eine Exception:

Hier scheint was mit dem PREG nicht zu stimmen, leider finde ich nicht
warum das nun auf einmal falsch sein soll.

Bsp: var_dump von result = string(126) " # # Table Data for
`fusion_smileys` # INSERT INTO `fusion_smileys` (`,`, `,`) VALUES
('1', ':)', 'smile.gif', 'Smile')"

Das ist kein korrekter query in Zeile 167 db_backup.php

iPaat commented:
Wenn ich das richtig gelesen habe, dann passiert das beim Restore?

Was steht denn in der Datei zu den Smileys?

globeFrEak commented:
ja genau beim Restore.

hier die Datei:
[backup.txt](https://github.com/PHP-Fusion-GERMANY/PHP-Fusion/files/336831/backup.txt)

iPaat commented:
Ich glaube der Fehler liegt hier:
https://github.com/globeFrEak/PHP-Fusion/blob/51cea07603f21ccb6be2baf440850303b0401516/administration/db_backup.php#L56-64

speziell in: `$field_name['name']` diese Variable ist leer.

globeFrEak committed:
fix Backup & Restore function

globeFrEak commented:
so geht es - ich musste ein wenig in den aktuellen php-fusion fork schauen :disappointed:

globeFrEak committed:
correct PDO close connection

iPaat commented:
Ist damit die Datenbank umgestellt?

globeFrEak commented:
Von meiner Seite ja, ich denke ich habe alle MySQL Zeilen umgeschrieben

iPaat commented:
Also wenn sich dann niemand mehr dazu meldet dann werde ich am Montag, den 4.7.2016 mergen.

iPaat commented:
Vielen Dank für deine Arbeit :+1:
